docs/source/docs/registry.md

```diff
@@ -1,32 +1,83 @@
 # Registry
 
-> This is a functionality under active development. A full documentation will be
-> released later.
-
-DeepReg is adopting the usage of the registry system recently, to facilitate the usage
-of custom functionalities including
-
-- neural network architecture
-- loss
-- optimizer
-- data pre-processing
-- data loader
-
-The registry is defined in `deepreg/registry.py`, where the class `Registry` maintains a
-dictionary mapping `(category, key)` to `value`. It also provides the
-`build_from_config` functionality, which allows creating a class instance from a config
-directly. This allows for the simplification of the config file such that each
-configuration only needs to provide the name of the class and the arguments to build the
-class for use.
-
-With the registry, when developing new classes **inside DeepReg**, we should use the
-corresponding decorator, so that the class is registered. Please check
-`deepreg/model/backbone/u_net.py` as an example. Moreover, the corresponding
-`__init__.py` (maybe multiple ones) should be modified so that these classes will be
-automatically registered when executing `import deepreg`. For defining custom classes
-outside DeepReg, more detailed tutorial will be released later.
-
-For now, we only support custom classes of
-
-- backbone
-- loss
+DeepReg adopts the
+[registry system](https://github.com/DeepRegNet/DeepReg/blob/main/deepreg/registry.py)
+to facilitate the addition of custom functionalities.
+
+## Description
+
+The class `Registry` maintains a dictionary mapping `(category, key)` to `value`, where
+
+- `category` is the class category, e.g. `"backbone_class""` for backbone classes.
+- `key` is the name of the registered class, e.g. `"unet"` for the class `UNet`.
+- `value` is the registered class, e.g. `UNet` corresponding to `"unet"`.
+
+A global variable `REGISTRY = Registry()` is defined to provide a central control of all
+classes. The supported categories and the registered classes are resumed in the
+[registered classes](registered_classes.html) page.
+
+### Register a class
+
+To register a class into `REGISTRY`, it is recommended to use `register` as a
+**decorator**. Consider the `UNet` class for backbone as an example.
+
+```python
+@REGISTRY.register(category="backbone_class", name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+The decorator will automatically register the class when this class get imported.
+Therefore, in order to make sure this class registered when we `import deepreg`, this
+class and related parent modules needs to be imported in the `__init__.py` files.
```

Comment:
Consider changing to: 'The decorator automatically registers the class upon import. To ensure that this class is registered when `import deepreg` is called, this class and related parent modules need to be imported in the `__init__.py` files.'